src/loaders/OmeZarrLoader.ts

```diff
@@ -496,7 +505,12 @@ class OMEZarrLoader extends ThreadableVolumeLoader {
       try {
         const result = await zarrGet(level, sliceSpec, { opts: { subscriber, reportKey } });
         const u8 = convertChannel(result.data);
-        onData(ch, u8);
+        if (this.syncChannels) {
+          resultChannelData.push(u8);
+          resultChannelIndices.push(ch);
+        } else {
+          onData([ch], [u8]);
+        }
```

Comment:
key change above: if syncChannels, then don't call onData but instead gather them into an array to call onData later after we are sure they have all arrived.